lib/ui/painting/image_encoding_impeller.cc

```diff
@@ -50,7 +54,23 @@ sk_sp<SkImage> ConvertBufferToSkImage(
   return raster_image;
 }
 
-void ConvertDlImageImpellerToSkImage(
+void DoConvertImageToRasterImpeller(
+    const sk_sp<DlImage>& dl_image,
+    std::function<void(sk_sp<SkImage>)> encode_task,
+    const std::shared_ptr<const fml::SyncSwitch>& is_gpu_disabled_sync_switch,
+    const std::shared_ptr<impeller::Context>& impeller_context) {
+  is_gpu_disabled_sync_switch->Execute(
+      fml::SyncSwitch::Handlers()
+          .SetIfTrue([&encode_task] { encode_task(nullptr); })
+          .SetIfFalse([&dl_image, &encode_task, &impeller_context] {
+            ImageEncodingImpeller::ConvertDlImageToSkImage(
+                dl_image, std::move(encode_task), impeller_context);
+          }));
+}
+
+}  // namespace
+
+void ImageEncodingImpeller::ConvertDlImageToSkImage(
```

Comment:
I made this public for easier testing.  With a little bit of work I could make the tests work on the existing public api.  There is just a bit of threading work required to do that.